fix: Add new doctypes for Contacts (#131)
cozy/cozy-doctypes: docs/io.cozy.contacts.md

The `io.cozy.contacts` doctype is loosely based on the [vCard RFC](https://tools.ietf.org/html/rfc6350), in that some of the attributes have been renamed for clarity. The attributes with a `?` are optional.

> - `index` : {string} Field to be used to sort contacts. It's a concatenation of `givenName`, `familyName`, first `email` and `cozy.url`

Reviewer: I don't think `index` is a good name, it's going to be confusing to set and index on `index`. An it's just one of possible indexes.

How about `sort` or `sortField` ?

Author: yes I proposed `sortBy`too. Ok for `sort` 👍

Reviewer 2: I'm wondering if we should not create something more robust over time. 

I don't know exactly yet, but having something like : 
```
indexes: {
 byGivenNameFamilyNameEmailCozyUrl: value
}
```
seems better to me.

Author: Est-ce qu'on commencerait pas par faire simple, pour complexifier ensuite si besoin ? (lean)

Reviewer 2: It's always a bit of work to migrate datas... We need to think of a fairly generic structure right from the start.

Reviewer: In that case, maybe `alphabeticalSort` ? It's more specific than `sort` and leaves us some options if we want other sorts, without being overly specific

Reviewer 2: Since it's not a "primary" data, I tend to think that we should not put it in the root of the document. Having an `indexes` object / array looks like a good idea. 

And since this entry is pretty special, having its name speaking for itself is mandatory. `alphabeticalSort ByGivenNameFamilyNameEmailCozyUrl` ?

Author: Is `alphabeticalSort` relevant here ? I mean, this will be part of the query (how to sort), not of how we create and store the value of this attribute (what to store).

And to create its value, we decided to concatenate givenName + familyName + mail + url but is it not simpler to get the first not empty of those fields ?

So go for this way : 
```
root {
   indexes: {
      ByGivenNameFamilyNameEmailCozyUrl: value
   }
}
```
?

Author: File change according to comments above

Reviewer: Good point on the alphabetical sort not being too relevant here. 👍 for the suggestion above.

> - `displayName` : {string} Name to be displayed. In order of priority, it's the `givenName` or `familyName` or first `email` or `cozy.url`

Reviewer: ```suggestion
- `displayName` : {string} Name to be displayed. In order of priority, it's the name, the email or the cozy URL
```

I don't think it's going to be givenName *or* familyName, we'll want both.

Author: It's `familyName` if `givenName` is empty. Actually it's `fullname` or first `email` or `cozy.url`

Reviewer: I still find the description not very clear and probably too technical. Even my proposal is not that good in hindsight, I don't think we should talk about the implementation details in this doc. We may want to change the rules without changing the meaning of the field.

Author: Ok good point 👍

Reviewer: Yeah we'll end up using it in drive as well. New description is good 👍  thanks

The `io.cozy.contacts` doctype is loosely based on the [vCard RFC](https://tools.ietf.org/html/rfc6350), in that some of the attributes have been renamed for clarity. The attributes with a `?` are optional.

> - `indexes` : {object} To be used to sort contacts by different ways

Reviewer: ```suggestion
- `indexes` : {object} Used to sort contacts in different ways
```

Author: Done 👍
